dask/dataframe/io.py

```diff
@@ -25,6 +25,30 @@
 
 csv_defaults = {'compression': None}
 
+
+def try_pd_read_csv(*args, **kwargs):
+    msg = """
+    By inspecting the first 1,000 rows of your csv file we guessed that the
+    column: %s had dtype: %s. But now that we're reading all the data we see it
+    has some values with dtype: %s. You should probably add the following
+    keyword arguments to your `read_csv` call:
+
+        dtype={'%s': %s}
+    """
+
+    try:
+        return pd.read_csv(*args, **kwargs)
+    except ValueError as e:
+        # this is brittle
```

Comment:
I have to parse the exception message to get the column name and dtypes. This is brittle and will break if the pandas exception message changes.